orchagent/neighorch.cpp

```diff
@@ -1755,12 +1759,18 @@ void NeighOrch::updateSrv6Nexthop(const NextHopKey &nh, const sai_object_id_t &n
         m_syncdNextHops.erase(nh);
     }
 }
-void NeighOrch::addZeroMacTunnelRoute(const NeighborEntry& entry, const MacAddress& mac)
+
+bool NeighOrch::addZeroMacTunnelRoute(const NeighborEntry& entry, const MacAddress& mac)
 {
     SWSS_LOG_INFO("Creating tunnel route for neighbor %s", entry.ip_address.to_string().c_str());
     MuxOrch* mux_orch = gDirectory.get<MuxOrch*>();
     NeighborUpdate update = {entry, mac, true};
     mux_orch->update(SUBJECT_TYPE_NEIGH_CHANGE, static_cast<void *>(&update));
-    m_syncdNeighbors[entry] = { mac, false };
-}
+    if (mux_orch->isStandaloneTunnelRouteInstalled(entry.ip_address))
```

Comment:
Shouldn't this check be moved to the beginning of the function? What would happen when calling mux_orch->update() when the tunnel routes is already installed?

Reply:
There's a check in the mux_orch->update() path that checks for this so it should be safe
```
if (update.add)
{
    if (standalone_tunnel_neigh_it == standalone_tunnel_neighbors_.end())
    {
        createStandaloneTunnelRoute(update.entry.ip_address);
    }
    /* If the MAC address in the neighbor entry is zero but the neighbor IP
     * is already present in standalone_tunnel_neighbors_, assume we have already
     * added a tunnel route for it and exit early
     */
    return;
}
```

Reply:
The check should be at the end. Imagine the tunnel route is first added.